cmd/agent/app/start.go

```diff
@@ -233,19 +242,29 @@ func StartAgent() error {
 
 // setupMetadataCollection initializes the metadata scheduler and its collectors based on the config
 func setupMetadataCollection(s *serializer.Serializer, hostname string) error {
-	addDefaultResourcesCollector := true
+	type collector struct {
+		os       string
+		interval time.Duration
+	}
+
+	// default collectors by os
+	defaultCollectors := map[string]collector{
+		"host":         collector{os: "*", interval: hostMetadataCollectorInterval},
+		"agent_checks": collector{os: "*", interval: agentChecksMetadataCollectorInterval},
+		"resources":    collector{os: "linux", interval: resourcesMetadataCollectorInterval},
```

Comment:
since `interval` is of type `time.Duration`, the conversion from `int` (in seconds) to `time.Duration` should be done here instead of when you call `common.MetadataScheduler.AddCollector`

(now that I write this, I realize the same mistake is present on the `MetadataProviders` struct definition. We should use `int` in that struct to match the config, so maybe for consistency the type of `collector.interval` defined in this function should be `int` as well)

Reply:
(this comment also applies to the dogstatsd file)